v2/blockstore/readonly.go

```diff
@@ -22,28 +22,38 @@ import (
 	internalio "github.com/ipld/go-car/v2/internal/io"
 )
 
-var _ blockstore.Blockstore = (*ReadOnly)(nil)
-
-// ReadOnly provides a read-only CAR Block Store.
-type ReadOnly struct {
-	// mu allows ReadWrite to be safe for concurrent use.
-	// It's in ReadOnly so that read operations also grab read locks,
-	// given that ReadWrite embeds ReadOnly for methods like Get and Has.
-	//
-	// The main fields guarded by the mutex are the index and the underlying writers.
-	// For simplicity, the entirety of the blockstore methods grab the mutex.
-	mu sync.RWMutex
-
-	// The backing containing the data payload in CARv1 format.
-	backing io.ReaderAt
-	// The CARv1 content index.
-	idx index.Index
-
-	// If we called carv2.NewReaderMmap, remember to close it too.
-	carv2Closer io.Closer
-
-	ropts carv2.ReadOptions
-}
+var (
+	_                    blockstore.Blockstore = (*ReadOnly)(nil)
+	noopErrHandler                             = func(error) {}
+	errZeroLengthSection                       = fmt.Errorf("zero-length section not allowed by default; see WithZeroLengthSectionAsEOF option")
```

Comment:
IMHO this grouping isn't great, as it adds a lot of aligning whitespace, and now the Errorf barely fits on most screens.

I'd add an empty line in between the first and second declarations, or just drop the grouping, as it's not gaining you much.

Reply:
`gofumpt` for you.

Reply:
I am curious what algorithm should I follow for grouping if `gofumpt` is not always perfect?

Reply:
this is gofmt; it will generally attempt to do vertical alignment if you group things together without empty lines. in this case, the grouping+alignment isn't gaining you much.